6.4.0 nvidia   thread #1 failed with error Unsupported algorithm

mining kawpow on NVIDIA GTX 1070 - works ok on xmrig 6.3.5, and stop working with 6.4.0
On AMD (RX 5500 and 5600) - work ok with both versions (xmrig using opencl)

[2020-10-19 09:00:57.681]  nvidia   KawPow DAG for epoch 194 calculated (14550ms)
[2020-10-19 09:00:57.681]  nvidia   thread #0 failed with error Unsupported algorithm
...and same error on all threads



Rig is Ubuntu based. Manually compiled with same procedure as always and on same machine as always:
xmrig-cuda:
cd cmake && export PATH=/usr/local/cuda-10.0/bin:$PATH; cmake .. -DCMAKE_C_COMPILER=gcc-5 -DCMAKE_CXX_COMPILER=g++-5 -DCUDA_LIB=/usr/local/cuda/lib64/stubs/libcuda.so && make -j12
xmrig:
cd cmake && cmake .. -DCMAKE_C_COMPILER=gcc-5 -DCMAKE_CXX_COMPILER=g++-5 && make -j32

running options:
--opencl --cuda --no-cpu -o stratum.ravenminer.com:3838 -a kawpow -u xxxxxxxxx -p x --print-time=5 --http-port=3333

full logs running 6.3.5 and 6.4.0 on this NV rig:

6.3.5:
 * ABOUT        XMRig/6.3.5 gcc/5.4.0
 * LIBS         libuv/1.8.0 OpenSSL/1.0.2g hwloc/1.11.9
 * HUGE PAGES   supported
 * 1GB PAGES    disabled
 * CPU          Intel(R) Celeron(R) CPU G3930 @ 2.90GHz (1) x64 AES
                L2:0.5 MB L3:2.0 MB 2C/2T NUMA:1
 * MEMORY       1.3/7.7 GB (16%)
 * DONATE       1%
 * ASSEMBLY     auto:intel
 * POOL #1      stratum.ravenminer.com:3838 algo kawpow
 * COMMANDS     hashrate, pause, resume, results, connection
 * HTTP API     127.0.0.1:3333
 * OPENCL       disabled (no devices)
 * CUDA         10.0/11.1/6.3.2
 * NVML         11.455.28/455.28 press e for health report
 * CUDA GPU     #0 01:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #1 03:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #2 06:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #3 07:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #4 08:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #5 09:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
[2020-10-19 08:58:53.020]  net      use pool stratum.ravenminer.com:3838  35.157.118.228
[2020-10-19 08:58:53.095]  net      new job from stratum.ravenminer.com:3838 diff 431M algo kawpow height 1457046
[2020-10-19 08:58:53.095]  nvidia   use profile  kawpow  (6 threads) scratchpad 32 KB
|  # | GPU |  BUS ID | INTENSITY | THREADS | BLOCKS | BF |  BS | MEMORY | NAME
|  0 |   0 | 01:00.0 |   7864320 |     256 |  30720 |  0 |   0 |   2576 | GeForce GTX 1070
|  1 |   1 | 03:00.0 |   7864320 |     256 |  30720 |  0 |   0 |   2576 | GeForce GTX 1070
|  2 |   2 | 06:00.0 |   7864320 |     256 |  30720 |  0 |   0 |   2576 | GeForce GTX 1070
|  3 |   3 | 07:00.0 |   7864320 |     256 |  30720 |  0 |   0 |   2576 | GeForce GTX 1070
|  4 |   4 | 08:00.0 |   7864320 |     256 |  30720 |  0 |   0 |   2576 | GeForce GTX 1070
|  5 |   5 | 09:00.0 |   7864320 |     256 |  30720 |  0 |   0 |   2576 | GeForce GTX 1070
[2020-10-19 08:58:57.188]  nvidia   READY threads 6/6 (4092 ms)
[2020-10-19 08:58:58.142]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:03.147]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:08.157]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:10.884]  miner    KawPow light cache for epoch 194 calculated (13696ms)
[2020-10-19 08:59:13.161]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:18.166]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:23.168]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:25.851]  nvidia   KawPow DAG for epoch 194 calculated (14966ms)
[2020-10-19 08:59:25.852]  nvidia   KawPow DAG for epoch 194 calculated (14967ms)
[2020-10-19 08:59:25.853]  nvidia   KawPow DAG for epoch 194 calculated (14968ms)
[2020-10-19 08:59:25.855]  nvidia   KawPow DAG for epoch 194 calculated (14970ms)
[2020-10-19 08:59:25.856]  nvidia   KawPow DAG for epoch 194 calculated (14971ms)
[2020-10-19 08:59:25.857]  nvidia   KawPow DAG for epoch 194 calculated (14972ms)
[2020-10-19 08:59:28.187]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:29.905]  net      new job from stratum.ravenminer.com:3838 diff 431M algo kawpow height 1457047
[2020-10-19 08:59:33.190]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 08:59:38.194]  miner    speed 10s/60s/15m 66.91 n/a n/a MH/s max 66.96 MH/s
[2020-10-19 08:59:38.979]  nvidia   accepted (1/0) diff 431M (122 ms)
[2020-10-19 08:59:43.197]  miner    speed 10s/60s/15m 67.21 n/a n/a MH/s max 67.22 MH/s
[2020-10-19 08:59:48.202]  miner    speed 10s/60s/15m 67.19 n/a n/a MH/s max 67.22 MH/s

6.4.0:
 * ABOUT        XMRig/6.4.0 gcc/5.4.0
 * LIBS         libuv/1.8.0 OpenSSL/1.0.2g hwloc/1.11.9
 * HUGE PAGES   supported
 * 1GB PAGES    disabled
 * CPU          Intel(R) Celeron(R) CPU G3930 @ 2.90GHz (1) x64 AES
                L2:0.5 MB L3:2.0 MB 2C/2T NUMA:1
 * MEMORY       1.3/7.7 GB (17%)
 * DONATE       1%
 * ASSEMBLY     auto:intel
 * POOL #1      stratum.ravenminer.com:3838 algo kawpow
 * COMMANDS     hashrate, pause, resume, results, connection
 * HTTP API     127.0.0.1:3333
 * OPENCL       disabled (no devices)
 * CUDA         10.0/11.1/6.4.0
 * NVML         11.455.28/455.28 press e for health report
 * CUDA GPU     #0 01:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #1 03:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #2 06:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #3 07:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #4 08:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
 * CUDA GPU     #5 09:00.0 GeForce GTX 1070 1771/4004 MHz smx:15 arch:61 mem:8022/8119 MB
[2020-10-19 09:00:25.048]  net      use pool stratum.ravenminer.com:3838  3.122.71.171
[2020-10-19 09:00:25.118]  net      new job from stratum.ravenminer.com:3838 diff 431M algo kawpow height 1457047
[2020-10-19 09:00:25.118]  nvidia   use profile  kawpow  (6 threads) scratchpad 32 KB
|  # | GPU |  BUS ID | INTENSITY | THREADS | BLOCKS | BF |  BS | MEMORY | NAME
|  0 |   0 | 01:00.0 |      5760 |     128 |     45 |  0 |   0 |   2576 | GeForce GTX 1070
|  1 |   1 | 03:00.0 |      5760 |     128 |     45 |  0 |   0 |   2576 | GeForce GTX 1070
|  2 |   2 | 06:00.0 |      5760 |     128 |     45 |  0 |   0 |   2576 | GeForce GTX 1070
|  3 |   3 | 07:00.0 |      5760 |     128 |     45 |  0 |   0 |   2576 | GeForce GTX 1070
|  4 |   4 | 08:00.0 |      5760 |     128 |     45 |  0 |   0 |   2576 | GeForce GTX 1070
|  5 |   5 | 09:00.0 |      5760 |     128 |     45 |  0 |   0 |   2576 | GeForce GTX 1070
[2020-10-19 09:00:29.169]  nvidia   READY threads 6/6 (4051 ms)
[2020-10-19 09:00:30.131]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:00:35.136]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:00:40.165]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:00:43.130]  miner    KawPow light cache for epoch 194 calculated (13961ms)
[2020-10-19 09:00:45.179]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:00:50.182]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:00:55.187]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:00:57.679]  nvidia   KawPow DAG for epoch 194 calculated (14549ms)
[2020-10-19 09:00:57.680]  nvidia   thread #1 failed with error Unsupported algorithm
[2020-10-19 09:00:57.681]  nvidia   KawPow DAG for epoch 194 calculated (14550ms)
[2020-10-19 09:00:57.681]  nvidia   thread #0 failed with error Unsupported algorithm
[2020-10-19 09:00:57.681]  nvidia   KawPow DAG for epoch 194 calculated (14551ms)
[2020-10-19 09:00:57.681]  nvidia   thread #3 failed with error Unsupported algorithm
[2020-10-19 09:00:57.682]  nvidia   KawPow DAG for epoch 194 calculated (14552ms)
[2020-10-19 09:00:57.684]  nvidia   thread #4 failed with error Unsupported algorithm
[2020-10-19 09:00:57.684]  nvidia   KawPow DAG for epoch 194 calculated (14554ms)
[2020-10-19 09:00:57.684]  nvidia   thread #2 failed with error Unsupported algorithm
[2020-10-19 09:00:57.684]  nvidia   KawPow DAG for epoch 194 calculated (14553ms)
[2020-10-19 09:00:57.684]  nvidia   thread #5 failed with error Unsupported algorithm
[2020-10-19 09:01:00.191]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:01:05.196]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2020-10-19 09:01:10.200]  miner    speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s

Hi
Using old 6.3.2 cuda plugin helps.

But the 6.4.0...
In the downloaded sources there is already set ON as default:
option(WITH_DRIVER_API "Enable CUDA Driver API and NVRTC, required for cn/r and kawpow algorithms" ON)

adding -DWITH_DRIVER_API=ON to cmake didnt help - same error during mining.
I even try to remove all "if" lines in sources, as below:
if (WITH_DRIVER_API)
in these few files...but still same error during mining.

Compile log below:
as I see the kawpow "stuff" is compiled.
maybe some other missing code is some there??

/xxx/xmrig-cuda-6.4.0# cuda 9.2; cd cmake && export PATH=/usr/local/cuda-9.2/bin:$PATH; cmake .. -DWITH_DRIVER_API=ON -DCMAKE_C_COMPILER=gcc-5 -DCMAKE_CXX_COMPILER=g++-5 -DCUDA_LIB=/usr/local/cuda/lib64/stubs/libcuda.so && make -j12
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/gcc-5
-- Check for working C compiler: /usr/bin/gcc-5 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++-5
-- Check for working CXX compiler: /usr/bin/g++-5 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for __builtin___clear_cache
-- Looking for __builtin___clear_cache - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found CUDA: /usr/local/cuda-9.2 (found suitable version "9.2", minimum required is "9.0")
-- Configuring done
-- Generating done
-- Build files have been written to: /home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/cmake
[  6%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/RandomX/arqma/xmrig-cu_generated_randomx_arqma.cu.o
[ 18%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/RandomX/keva/xmrig-cu_generated_randomx_keva.cu.o
[ 18%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/RandomX/monero/xmrig-cu_generated_randomx_monero.cu.o
[ 25%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/xmrig-cu_generated_cuda_extra.cu.o
[ 31%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/KawPow/raven/xmrig-cu_generated_KawPow.cu.o
[ 37%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/RandomX/wownero/xmrig-cu_generated_randomx_wownero.cu.o
[ 43%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/xmrig-cu_generated_cuda_core.cu.o
[ 50%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/AstroBWT/dero/xmrig-cu_generated_AstroBWT.cu.o
[ 56%] Building NVCC (Device) object CMakeFiles/xmrig-cu.dir/src/RandomX/xmrig-cu_generated_randomx.cu.o
/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/RandomX/randomx_cuda.hpp(319): warning: pointless comparison of unsigned integer with zero

/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/RandomX/randomx_cuda.hpp(319): warning: pointless comparison of unsigned integer with zero

/usr/local/cuda-9.2/include/thrust/system/cuda/detail/util.h:189:13: warning: ‘void thrust::cuda_cub::throw_on_error(cudaError_t, const char*)’ defined but not used [-Wunused-function]
 throw_on_error(cudaError_t status, char const *msg)
             ^
cc1plus: warning: unrecognized command line option ‘-Wno-class-memaccess’
/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/RandomX/randomx_cuda.hpp(319): warning: pointless comparison of unsigned integer with zero

/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/RandomX/randomx_cuda.hpp(319): warning: pointless comparison of unsigned integer with zero

/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/RandomX/randomx_cuda.hpp(319): warning: pointless comparison of unsigned integer with zero

/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/cuda_extra.cu: In function ‘int cuda_get_deviceinfo(nvid_ctx*)’:
/home/miner/tym/xmrig-cuda/xmrig-cuda-6.4.0/src/cuda_extra.cu:569:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (ctx->device_blocks > max_blocks) {
                          ^
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-class-memaccess’
Scanning dependencies of target xmrig-cu
[ 62%] Building CXX object CMakeFiles/xmrig-cu.dir/src/KawPow/raven/CudaKawPow_gen.cpp.o
[ 68%] Linking CXX static library libxmrig-cu.a
[ 68%] Built target xmrig-cu
Scanning dependencies of target xmrig-cuda
[ 75%] Building C object CMakeFiles/xmrig-cuda.dir/src/crypto/cn/c_blake256.c.o
[ 81%] Building CXX object CMakeFiles/xmrig-cuda.dir/src/crypto/common/Algorithm.cpp.o
[ 87%] Building CXX object CMakeFiles/xmrig-cuda.dir/src/xmrig-cuda.cpp.o
[ 93%] Building CXX object CMakeFiles/xmrig-cuda.dir/src/CudaCryptonightR_gen.cpp.o
[100%] Linking CXX shared library libxmrig-cuda.so
[100%] Built target xmrig-cuda

--- a/src/crypto/common/Algorithm.h
+++ b/src/crypto/common/Algorithm.h
@@ -64,6 +64,7 @@ public:
         RX_SFX,        // "rx/sfx"           RandomSFX (Safex Cash).
         RX_KEVA,       // "rx/keva"          RandomKV (Keva).
         AR2_CHUKWA,    // "argon2/chukwa"    Argon2id (Chukwa).
+        AR2_CHUKWA_V2, // "argon2/chukwav2"  Argon2id (Chukwa v2).
         AR2_WRKZ,      // "argon2/wrkz"      Argon2id (WRKZ)
         ASTROBWT_DERO, // "astrobwt"         AstroBWT (Dero)
         KAWPOW_RVN,    // "kawpow/rvn"       KawPow (RVN)
@@ -172,6 +173,9 @@ public:
             case AR2_CHUKWA:
                 return oneMiB / 2;
 
+            case AR2_CHUKWA_V2:
+                return oneMiB;
+
             case AR2_WRKZ:
                 return oneMiB / 4;
 
@@ -235,6 +239,7 @@ public:
             return RANDOM_X;
 
         case AR2_CHUKWA:
+        case AR2_CHUKWA_V2:
         case AR2_WRKZ:
             return ARGON2;
 

